### ISSUE TITLE
does this show changes/blood flow/heartbeat?

### ISSUE THREAD

**Issue by atv2016** [kgram007/Eulerian-Motion-Magnification#1]

**Comment by kgram007** [comment]
Yes, in principle it should be able to magnify blood flow and heartbeats. Take a look at this video: https://www.youtube.com/watch?v=e9ASH8IBJ2U

For further details, refer to the paper: http://people.csail.mit.edu/mrub/papers/vidmag.pdf

**Comment by atv2016** [comment]
Hi,
Thanks for your reply. You did some great work here, many thanks for letting people take a look. It’s really quite cool stuff.

A few questions:
-Could one enlarge this even more? Increasing the scale factor doesn’t seem to make a difference.
-How would one uses this to read a heartbeat ?

Thanks,
Alef
On 21 Dec 2016, at 19:47, kgram007 <notifications@github.com<mailto:notifications@github.com>> wrote:


Yes, in principle it should be able to magnify blood flow and heartbeats. Take a look at this video: https://www.youtube.com/watch?v=e9ASH8IBJ2U

For further details, refer to the paper: http://people.csail.mit.edu/mrub/papers/vidmag.pdf

**Comment by atv2016** [comment]
I wanted to use this, but it uses GSL (gnu scientific libraries) and i can't use that. Do you see any other way of doing this? I really appreciate your time and help.

**Comment by kgram007** [comment]
Hi,
Its been sometime since I last modified the code. Also some of the functions that I had used from OpenCV have now been deprecated. So, I will be refactoring the code and add CMakeList.txt file for easy compiling. I will complete the modification and push it to the repo.

With the modification, I guess it should be easy to run the program. Also I will then look into testing the code with multiple alpha values and let you know which is best params for detecting heartbeats.

Regarding the GSL, which part of code/function are you talking about?

**Comment by atv2016** [comment]
Sorry i meant, i wanted to use this:https://github.com/melfm/pulse-detector

And apply the same logic/code, but it uses GSL and i'm not sure how easy it would be to convert it.
I think i had to modify some things but it compiles fine for me otherwise. But yeah it would be great if we could add heartbeat.

Alef

**Comment by atv2016** [comment]
Also: the video writer seems to write until the disk is full :-)

**Comment by atv2016** [comment]
Hi,
I just noticed, that it runs fine without writing a video to file. I can’t quite see the purpose of this, am i missing s something ?

Thanks!
On 22 Dec 2016, at 08:47, kgram007 <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi,
Its been sometime since I last modified the code. Also some of the functions that I had used from OpenCV have now been deprecated. So, I will be refactoring the code and add CMakeList.txt file for easy compiling. I will complete the modification and push it to the repo.

With the modification, I guess it should be easy to run the program. Also I will then look into testing the code with multiple alpha values and let you know which is best params for detecting heartbeats.

Regarding the GSL, which part of code/function are you talking about?
